src/Interpreters/TranslateQualifiedNamesVisitor.cpp

```diff
@@ -29,38 +31,56 @@ namespace ErrorCodes
     extern const int UNSUPPORTED_JOIN_KEYS;
     extern const int LOGICAL_ERROR;
 }
+bool TranslateQualifiedNamesMatcher::Data::matchColumnName(const String & name, const String & column_name, DataTypePtr column_type)
```

Comment:
Maybe consider use of std::string_view here?
```suggestion
bool TranslateQualifiedNamesMatcher::Data::matchColumnName(const std::string_view & name, const String & column_name, DataTypePtr column_type)
```